test/functional/p2p_blocksonly.py

```diff
@@ -78,6 +79,34 @@ def blocksonly_mode_tests(self):
         self.nodes[0].disconnect_p2ps()
         self.generate(self.nodes[0], 1)
 
+    # Tests with an outbound peer that sets txrelay to false (which they do if they are in blocksonly mode)
+    def blocksonly_peer_tests(self):
+        self.restart_node(0, ["-noblocksonly"])  # disables blocks only mode
+        self.log.info("Check that we sustain full-relay outbound connections to blocksonly peers when not at the full outbound limit")
+        blocksonly_peer = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=1, txrelay=False)
+        # Trigger CheckForStaleTipAndEvictPeers, which is scheduled every 45s (EXTRA_PEER_CHECK_INTERVAL)
+        self.nodes[0].mockscheduler(46)
+        blocksonly_peer.sync_with_ping()
+        blocksonly_peer.peer_disconnect()
+
+        self.log.info("Check that we evict full-relay outbound connections to blocksonly peers at the full outbound limit")
+        # Simulate the situation of being at the max limit of full-outbound connections by setting -maxconnections to a low value
+        self.restart_node(0, ["-noblocksonly", "-maxconnections=2"])
+        # Have one blocksonly peer that is not being disconnected
+        blocksonly_peer1 = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=1, txrelay=False)
+        blocksonly_peer2 = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=2, txrelay=False)
+        self.nodes[0].mockscheduler(46)
```

Comment:
68769ff: nit if you retouch.
```suggestion
with self.nodes[0].assert_debug_log(expected_msgs=["disconnecting full outbound peer not participating in tx relay: peer=1"]):
            self.nodes[0].mockscheduler(46)
            blocksonly_peer2.wait_for_disconnect()
```